cocoapods-beta: init at 1.7.0.beta.3

lilyball opened the pull request:
###### Motivation for this change
cocoapods-beta is an alternative attribute for the cocoapods package that provides the latest beta instead of the stable version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

lilyball committed:
cocoapods-beta: init at 1.7.0.beta.3

cocoapods-beta is an alternative attribute for the cocoapods package
that provides the latest beta instead of the stable version.

Also switch to `bundlerApp` so we stop polluting the profile with our
gem dependencies and can get rid of the broken `sandbox-pod` executable.

matthiasbeyer reviewed (changes requested):
The commit message does not follow the guidelines. Sorry.

lilyball commented:
@matthiasbeyer Are you talking about the subject, or the body? I'm not aware of guidelines around the commit message body, and the subject should follow guidelines.

I know our guidelines say to use the package name, not the attribute name, but I went back and checked the commit that introduced `nodejs_11_x` (because that's the same scenario, where the package already exists but we're introducing a different attribute for a new version) and it did what I did here, which is use the attribute name with the `init` style.

nixos-discourse commented:
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/rubygem-conflicts-with-ruby-packages/2636/5

lilyball committed:
Make gemdir optional for bundlerApp

Like `bundlerEnv`, the `gemdir` parameter to `bundlerApp` can be omitted
if all 3 of `gemfile`, `lockfile`, and `gemset` are provided.

lilyball commented:
I've just updated this PR 2 with 2 changes:

- Switched to `bundlerApp` so we stop linking our dependencies into the profile, and can get rid of the broken `sandbox-pod` executable.
- Tweaked `bundlerApp` to make the `gemdir` parameter optional just like it is in `bundlerEnv`, as it's unnecessary if we're providing all 3 of `gemfile`, `lockfile`, and `gemset`.

matthiasbeyer commented:
The commit message is `init: cocoapods-beta at 1.7.0.beta.3` but should be `cocoapods-beta: init at 1.7.0.beta.3`.

alyssais commented on pkgs/development/mobile/cocoapods/default.nix:
```diff
@@ -1,9 +1,13 @@
-{ lib, bundlerEnv, ruby }:
+{ lib, bundlerApp, ruby,
+  beta ? false }:
```
Commas at the start of line, please.

alyssais reviewed (commented):
`bundlerApp` change looks good :)

lilyball commented:
@matthiasbeyer Oh oops, you're right. I'll fix it.

lilyball commented:
Both issues (commit message and comma) have been fixed.

alyssais commented on pkgs/development/mobile/cocoapods/update:
```diff
@@ -1,11 +1,15 @@
-rm -rf "$BUNDIX_CACHE"
-rm -rf .bundle
+bundler lock --update
+# For some reson `bundler lock` doesn't take a `--gemfile` flag
```
Does the `BUNDLE_GEMFILE` environment variable work?

lilyball replied:
I just tested, looks like the answer is "no". I have to delete Gemfile-beta.lock first as apparently `bundler lock --update` won't backdate a lockfile despite the documentation claiming that flag makes it ignore the current lockfile (🤦🏻‍♀️) but after doing that, passing `env BUNDLE_GEMFILE=Gemfile-beta bundler lock --update --lockfile=Gemfile-beta.lock`  produces a lockfile that's clearly coming from `Gemfile`.

alyssais replied:
Grr, okay. If you open a Bundler bug and add it to the comment, I'm happy to merge this.

lilyball replied:
Huh. I just tested again to make sure and it actually seems like it's working now. I don't know what I did wrong before, since I double-checked my work last time, but if it really does work, I'll update this script to use that.

alyssais replied:
Woo!

lilyball commented:
@alyssais PR updated to use `BUNDLE_GEMFILE` instead of the shell trickery.

alyssais commented:
Thanks @lilyball
